From 4294037a8d43eae7d141560733b8d07b9382cca1 Mon Sep 17 00:00:00 2001
From: Ecco park <eccopark@broadcom.com>
Date: Wed, 2 Jul 2014 16:59:33 -0700
Subject: [PATCH 24/77] net: wireless: bcmdhd: fix bug for RTT

1) event array should be stored in order
2) rtt_results->distance is set to -1 only for invalid value
Signed-off-by: Ecco park <eccopark@broadcom.com>
---
 drivers/net/wireless/bcmdhd/wl_cfgvendor.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/bcmdhd/wl_cfgvendor.c b/drivers/net/wireless/bcmdhd/wl_cfgvendor.c
index 189840dab654..7b8e4470c536 100644
--- a/drivers/net/wireless/bcmdhd/wl_cfgvendor.c
+++ b/drivers/net/wireless/bcmdhd/wl_cfgvendor.c
@@ -767,7 +767,7 @@ void wl_cfgvendor_rtt_evt(void *ctx, void *rtt_data)
 			rtt_report->rtt = rtt_result->meanrtt;
 			rtt_report->rtt_sd = rtt_result->sdrtt;
 			/* convert to centi meter */
-			if (rtt_result->distance == 0xffffffff)
+			if (rtt_result->distance != 0xffffffff)
 				rtt_report->distance = (rtt_result->distance >> 2) * 25;
 			else /* invalid distance */
 				rtt_report->distance = -1;
@@ -1099,11 +1099,13 @@ static const struct  nl80211_vendor_cmd_info wl_vendor_events [] = {
 		{ OUI_GOOGLE, GOOGLE_GSCAN_GEOFENCE_FOUND_EVENT },
 		{ OUI_GOOGLE, GOOGLE_GSCAN_BATCH_SCAN_EVENT },
 		{ OUI_GOOGLE, GOOGLE_SCAN_FULL_RESULTS_EVENT },
-		{ OUI_GOOGLE, GOOGLE_SCAN_COMPLETE_EVENT },
 #endif /* GSCAN_SUPPORT */
 #ifdef RTT_SUPPORT
 		{ OUI_GOOGLE, GOOGLE_RTT_COMPLETE_EVENT },
 #endif /* RTT_SUPPORT */
+#ifdef GSCAN_SUPPORT
+		{ OUI_GOOGLE, GOOGLE_SCAN_COMPLETE_EVENT },
+#endif /* GSCAN_SUPPORT */
 };
 
 int wl_cfgvendor_attach(struct wiphy *wiphy)
-- 
2.0.3

